keychain_linux.cpp

```diff
@@ -99,40 +100,44 @@ void setPassword(const std::string &package, const std::string &service,
                                user.c_str(),
                                NULL);
 
-    updateError(err, error);
+    if (error != NULL) {
```

Comment:
Either this, or the analog check in `updateError` is not really needed anymore. Do you think we should drop one of the two?
(Discussion applies to all API functions on all three platforms)

Reply:
I would vote for keeping both checks. Its redundant, yes. However, for consistency with the other methods, where you have these `if {} else` constructions, I think you should keep the extra check here. And inside `updateError()` you'd better check for `nullptr` before accessing the `GError` struct. The latter might be okay to become an `assertion` though?

Reply:
Cool :) Then I'll keep it as it is.
Re: assertion: Tim says he doesn't like assertions and we shouldn't make the user's code crash. Probably won't, but setting the Error to success in the success case also fits the name `updateError` better. I think...